src/app/stack/stack-details/stack-details.component.ts

```diff
@@ -227,7 +234,104 @@ export class StackDetailsComponent implements OnChanges {
         this.componentFilterBy = filterBy.filterBy;
     }
 
-    constructor(private stackAnalysisService: StackAnalysesService) { }
+    modalRef: BsModalRef;
+    token: string;
+    tokenerror: TokenErrorModel = {
+        status: false,
+        type: '',
+        length: 0
+    };
+    tokenErrorStatus: boolean = true;
+    tokenDetail: TokenDetailModel = {
+        id: '',
+        status: 'freetier'
+    }
+    tokenAlertsMessage: string;
+
+
+    observableToken: Observable<any>;
+
+    constructor(private stackAnalysisService: StackAnalysesService, private modalService: BsModalService) { }
+
+    async submitToken() {
+        if (!this.tokenErrorStatus) {
+            await this.stackAnalysisService.linkSynkTokenWithUserID(this.getBaseUrl(this.stack), this.uuid, this.token, this.gatewayConfig)
+                .then(res => {
+                    console.log(res.status);
+                })
+                .catch(err => {
+                    console.log(err);
+                });
+
+            console.log('refresh in 5 sec');
+            setTimeout(() => {
+                this.init()
+            }, 5000);
```

Comment:
What is the logic behind this 5sec?

Reply:
Its just a wait time of 5 sec. Just to achieve consistency on the getCall inside init().